Lib/TwitterAds/TwitterAdsAuthManager.php

```diff
@@ -0,0 +1,89 @@
+<?php
+App::uses('TwitterAdsAuthContainer','AuthManager.Lib/TwitterAds');
+App::uses('MediaPlatformAuthManager','AuthManager.Lib');
+
+/**
+ * Class TwitterAdsAuthManager
+ */
+class TwitterAdsAuthManager extends MediaPlatformAuthManager {
+
+	/**
+	 * @var \Codebird\Codebird
+	 */
+	private $codebird;
+
+	/**
+	 * TwitterAdsAuthManager constructor.
+	 */
+	public function __construct() {
+		parent::__construct();
+		Configure::load('AuthManager.API/TwitterAds');
+		\Codebird\Codebird::setConsumerKey(
+			Configure::read('TwitterAds.consumer_key'),
+			Configure::read('TwitterAds.consumer_secret')
+		);
+		$this->codebird = \Codebird\Codebird::getInstance();
+	}
+
+	/**
+	 * Get the authentication url to add an user.
+	 *
+	 * @return string
+	 */
+	public function getAuthUrl() {
+		$tokens = $this->codebird->oauth_requestToken([
+			'oauth_callback' => $this->_getCallbackUrl(MediaPlatform::TWITTER_ADS)
+		]);
+		CakeSession::write('oauth_token_secret', $tokens->oauth_token_secret);
```

Comment:
@ronoudgenoeg Yep, Google/FB/Bing Ads/noem het maar op gebruikt OAUTH2, Twitter gebruikt OAUTH1a wat net wat anders is..